### ISSUE TITLE
Invalid HTTP responses

### ISSUE THREAD

**Issue by adamtwiss** [flynn-archive/flynn-demo#31]
Hi,
Am not sure which flynn sub-project this should go in, but the HTTP validity of the HTTP 'load-balancer' isn't correct. See:

```
$ telnet localhost 8080
Connected to localhost.
Escape character is '^]'.
GET / HTTP/1.0
Host: localhost:8080

HTTP/1.1 200 OK
Transfer-Encoding: chunked
Connection: keep-alive
Content-Type: text/plain
Date: Fri, 25 Apr 2014 12:58:01 GMT

1a
Hello Flynn on port 55013

0
```

The chunked-transfer encoding was added in HTTP/1.1. It isn't valid to send such a response to an HTTP/1.0 request. Even with a 1.1 client you probably shouldn't send chunked requests without the presence of an Accept-Encoding: chunked

Refs:
Sections 3.6 of HTTP Spec: http://www.w3.org/Protocols/rfc2616/rfc2616-sec3.html
http://en.wikipedia.org/wiki/Chunked_transfer_encoding

**Comment by lmars** [comment]
This issue belongs in the Flynn router: https://github.com/flynn/strowger.

The router always uses HTTP 1.1: https://github.com/flynn/strowger/blob/master/http.go#L368

**Comment by titanous** [comment]
Yep, this is invalid in HTTP 1.0, we should definitely fix. I'm pretty sure all valid HTTP 1.1 clients need to handle chunked encoding though, do you have a reference for the `Accept-Encoding` part of this issue?

**Comment by adamtwiss** [comment]
Reading a bit more about it, and testing it.

Impact: The current behaviour breaks any HTTP/1.0 client. Almost every browser on any device is now HTTP/1.1 compliant, but some test programs, or API clients might only support HTTP/1.0

What's wrong:
We're sending back HTTP/1.1 requests with chunked encoding to an HTTP/1.0 client
HTTP/1.1 Spec section 3.6:  "A server MUST NOT send transfer-codings to an HTTP/1.0 client."

My comment above about requiring Accept-Encoding on an HTTP/1.1isn't valid, as an HTTP/1.1 client must support chunked encoding HTTP Spec 3.6.1: "All HTTP/1.1 applications MUST be able to receive and decode the "chunked" transfer-coding"

The fix is to to make sure that for HTTP/1.0 clients that we send back a non-chunk encoded response.

**Comment by titanous** [comment]
There are a couple more HTTP/1.0 behaviors that we need to do like closing connections after each request unless `Keep-Alive` is set.
